### PR TITLE
Support for user provided plugins

### PR DESCRIPTION
**Pull request by parauliya**
## Pull request description 
This PR is supposed to add a feature to support user provided plugins for Jmeter test

## Checklist (choose whats happened)

- [ ] breaking change! (describe)

- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: contrib/executor/jmeterd/scripts/jmeter-master.sh [hunk 1]
```diff
@@ -19,6 +19,21 @@ else
...
+
+if [ -d $JMETER_USER_PLUGINS_FOLDER ]
+then
+  echo "Installing custom plugins from ${JMETER_USER_PLUGINS_FOLDER}"
+  for plugin in ${JMETER_USER_PLUGINS_FOLDER}/*.jar; do
+      echo "Copying plugin $plugin to ${JMETER_HOME}/lib/ext/${plugin}"
+      cp $plugin ${JMETER_HOME}/lib/ext
+  done;
+else
+  echo "No custom plugins found in ${JMETER_USER_PLUGINS_FOLDER}"
```
Comment (sahil-lakhwani) [inline contrib/executor/jmeterd/scripts/jmeter-master.sh]: should be `No user plugins` ?

File: contrib/executor/jmeterd/scripts/jmeter-slaves.sh [hunk 2]
```diff
@@ -19,6 +19,20 @@ else
...
 
+if [ -d $JMETER_USER_PLUGINS_FOLDER ]
+then
+  echo "Installing custom plugins from ${JMETER_USER_PLUGINS_FOLDER}"
+  for plugin in ${JMETER_USER_PLUGINS_FOLDER}/*.jar; do
+      echo "Copying plugin $plugin to ${JMETER_HOME}/lib/ext/${plugin}"
+      cp $plugin ${JMETER_HOME}/lib/ext
+  done;
+else
+  echo "No custom plugins found in ${JMETER_USER_PLUGINS_FOLDER}"
```
Comment (sahil-lakhwani) [inline contrib/executor/jmeterd/scripts/jmeter-slaves.sh]: should be `No user plugins`?

File: contrib/executor/jmeterd/pkg/runner/runner.go [hunk 3]
```diff
@@ -115,6 +114,16 @@ func (r *JMeterRunner) Run(ctx context.Context, execution testkube.Execution) (r
...
 	}
 
+	pluginPath := filepath.Join(filepath.Dir(path), "plugins")
+	// Set env plugin env variable to set custom plugin directory
+	// with this path custom plugin will be copied to jmeter's plugin directory
+	err = os.Setenv("JMETER_USER_PLUGINS_FOLDER", pluginPath)
+	if err != nil {
+		output.PrintLogf("%s Failed to set custom plugin directory %s", ui.IconWarning, pluginPath)
+	}
+	// Add custom plgin folder in slaves env variables
```
Comment (sahil-lakhwani) [inline contrib/executor/jmeterd/pkg/runner/runner.go]: nit: `s/plgin/plugin`

File: contrib/executor/jmeterd/pkg/runner/runner.go [hunk 4]
```diff
@@ -115,6 +114,16 @@ func (r *JMeterRunner) Run(ctx context.Context, execution testkube.Execution) (r
...
+	slavesEnvVariables["JMETER_USER_PLUGINS_FOLDER"] = testkube.NewBasicVariable("JMETER_USER_PLUGINS_FOLDER", pluginPath)
+
```
Comment (sahil-lakhwani) [inline contrib/executor/jmeterd/pkg/runner/runner.go]: Do we need the below here, as it was in the previous commit?
```
envManager.Variables["JMETER_USER_PLUGINS_FOLDER"] = testkube.Variable{Name: "JMETER_USER_PLUGINS_FOLDER", Value: pluginPath}
```

**Comment by sahil-lakhwani** [comment]
Also. let's update the PR description please.

File: contrib/executor/jmeterd/pkg/runner/runner.go (hunk 4 above)
Comment (hiteshwani) [reply contrib/executor/jmeterd/pkg/runner/runner.go]: Yes it is required because we have different map ( slavesEnvVariables ) for slaves envs.

Comment (sahil-lakhwani) [reply contrib/executor/jmeterd/pkg/runner/runner.go]: That line is not in the latest commit, do we need to add it?

**Pull request merged**